diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/OrderedRuleSet.kt

```diff
@@ -1,28 +1,52 @@
-@file:Suppress(
-    "Deprecation"
-)
-
 package org.cqfn.diktat.ruleset.rules
 
 import org.cqfn.diktat.common.config.rules.qualifiedWithRuleSetId
 import org.cqfn.diktat.ruleset.constants.EmitType
 import com.pinterest.ktlint.core.Rule
-import com.pinterest.ktlint.core.RuleSet
+import com.pinterest.ktlint.core.RuleProvider
 import com.pinterest.ktlint.core.api.EditorConfigProperties
 import org.jetbrains.kotlin.com.intellij.lang.ASTNode
 
 /**
  * This is a wrapper around Ktlint RuleSet which adjusts visitorModifiers for all rules to keep order with prevRule
  * Added as a workaround after introducing a new logic for sorting KtLint Rules: https://github.com/pinterest/ktlint/issues/1478
  *
- * @param id ID of RuleSet
- * @param rules rules which belongs to current RuleSet
+ * @param rules the rules which belong to the current [DiktatRuleSet].
  */
-class OrderedRuleSet(id: String, vararg rules: Rule) : RuleSet(id, rules = adjustRules(id, rules = rules)) {
+class OrderedRuleSet private constructor(
```

Comment:
suggestion to combine DiktatRuleSetProviderV2 and OrderedRuleSet: Diktat's proxy to RuleSetProviderV2 should keep ordering

Reply:
It actually does keep the ordering -- see [`DiktatRuleSetProvider#invoke`, line 224](https://github.com/saveourtool/diktat/blob/feature/ktlint-wrapper/diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/DiktatRuleSetProvider.kt#L224).

It's actually the rule proxy you were talking about eralier: it encapsulates rule discovery, configuration and ordering while no longer implementing any _KtLint_-specific SPI interface (`RuleSetProviderV2` in our case).

My idea was to rename `DiktatRuleSetProvider` to `DiktatRuleSetFactoryImpl` or something: as you can see, this class now implements a functional `DiktatRuleSetFactory` interface which allows to decouple SPI implementation (`DiktatRuleSetProviderV2`) from Diktat-specific ordered rule set implementation (`DiktatRuleSetProvider` and `OrderedRuleSet`).

There's one issue with `OrderedRuleSet` implementaion, though: it still relies on `Rule.VisitorModifier` which is _KtLint_-specific API.

Reply:
We need to add `Rule.VisitorModifier` only when create `RuleSetProviderV2` and it only ktlint specific.
So suggestion is to contain all logic related to creating KtLintRule in a single place.
It can be done in phase 2 when we will remove direct usage of Ktlint Rule

Reply:
I've updated the corresponding [issue description](https://github.com/saveourtool/diktat/issues/1543#issue-1429691094).